packages/x-components/src/plugins/x-plugin.mixin.ts

```diff
@@ -15,8 +16,8 @@ declare module 'vue/types/vue' {
 }
 
 interface PrivateExtendedVueComponent extends Vue {
+  $location?: QueryLocation | ResultLocation;
```

Comment:
If we want this injected location to be reactive, we need the @XProvide and @XInject